### PR TITLE
add initial support for intel npu acceleration library

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Add initial support for intel npu acceleration library, just keep ipex-llm's API, calling `intel_npu_acceleration_library`'s `compile` function directly

### 2. User API changes

```python
from ipex_llm.transformers.npu_model import AutoModelForCausalLM
model = AutoModelForCausalLM.from_pretrained(model_path, trust_remote_code=True, load_in_low_bit='fp16')
```

`ipex_llm.transformers.npu_model.AutoModelForCausalLM` receives all arguments of `ipex_llm.transformers.AutoModelForCausalLM`, but it doesn't support most of them, so it just ignores them.

`load_in_low_bit` only supports `sym_int4`, `sym_int8` and `fp32`

**Note: import ipex will cause `intel_npu_acceleration_library` cannot find NPU, so we must disable ipex auto-import when use it by setting `set BIGDL_IMPORT_IPEX=0`**

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/npu_model.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        """
+        if kwargs.get('device_map', None) not in [None, 'cpu', 'auto']:
+            warnings.warn("`device_map` will be ignored")
+        kwargs['device_map'] = 'cpu'
+
+        low_bit = kwargs.pop('load_in_low_bit', None)
+        low_bit_to_dtype_map = {
+            'sym_int4': int4,
+            'sym_int8': int8,
+            'fp32': torch.float,
```
Comment (rnwang04) [inline python/llm/src/ipex_llm/transformers/npu_model.py:71]: how about fp16?

**Review by rnwang04 — APPROVED** [review]
others LGTM as an initial PR.

File: python/llm/src/ipex_llm/transformers/npu_model.py (hunk 1 above)
Comment (MeouSker77) [reply python/llm/src/ipex_llm/transformers/npu_model.py:71]: it only moves Linear Computation to NPU, other OPs are still run on CPU, CPU doesn't support most fp16 operations

File: python/llm/src/ipex_llm/transformers/npu_model.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                        *args,
+                        **kwargs):
+        """
+        Load a model from a directory or the HF Hub. Use load_in_low_bit parameter to convert
+        model to low-bit format, like int4 and int8.
+        The loaded model will run supported OPs on NPU, then run other OPs on CPU.
+
+        Three new arguments are added to extend Hugging Face's from_pretrained method as follows:
+        :param load_in_low_bit: str value, options are ``'sym_int4'``, ``'sym_int8'``,
+                                ``'fp16'`` and ``'fp32'``.
```
Comment (rnwang04) [inline python/llm/src/ipex_llm/transformers/npu_model.py:59]: So we should not declare `fp16` support here ?

**Comment by jason-dai** [comment]
> **Note: import ipex will cause `intel_npu_acceleration_library` cannot find NPU, so we must disable ipex auto-import when use it by setting `set BIGDL_IMPORT_IPEX=0`**

Maybe add another ENV "BIGDL_USE_NPU" to disable auto import

**Comment by MeouSker77** [comment]
> Maybe add another ENV "BIGDL_USE_NPU" to disable auto import

added it

**Review by jason-dai — APPROVED** [review]
LGTM

**Pull request merged**